pkg/sql/plan/base_binder.go

```diff
@@ -893,6 +893,17 @@ func bindFuncExprImplByPlanExpr(name string, args []*Expr) (*plan.Expr, error) {
 		if args[1].Typ.Id == int32(types.T_any) {
 			args[1].Typ.Id = int32(types.T_varchar)
 		}
+	case "timediff":
```

Comment:
i don't think this code was needed.
function register can do this work

Reply:
the function register will cast the type if the inputs type are different, but the timediff function don't allowed two  inputs with different types (like select timediff(CAST('2017-08-08 22:22:22' as datetime), cast('11:11:11' as time));)